src/crypto.js

```diff
@@ -40,12 +40,10 @@ exports.getRandomValues = function (buffer) {
     view = new Uint8Array(buffer),
     i;
 
-  if (offset + size > buf.length) {
-    throw new Error("Insufficient Randomness Allocated.");
-  }
-
+  // Hack: just wrap around.
```

Comment:
Reusing randomness is bad news bears. We need to figure out what part of algorithm is using more bytes of randomness than we expect, and instead request the correct amount of entropy from the system.
This approach absolutely opens the algorithm to confirmation / information disclosure attacks.